contrib/IECoreUSD/src/IECoreUSD/PrimitiveAlgo.cpp

```diff
@@ -295,6 +295,15 @@ bool readPrimitiveVariables( const pxr::UsdSkelRoot &skelRoot, const pxr::UsdGeo
 		return false;
 	}
 
+	// the UsdSkelSkinningQuery gives us the points in skeleton space, but we have
+	// computed the location transforms separately, so we transform the points by the
+	// inverse bind matrix to put them back into prim-local space.
+	pxr::GfMatrix4d inverseBind = skinningQuery.GetGeomBindTransform( time ).GetInverse();
+	for( auto &p : points )
+	{
+		p = inverseBind.Transform( p );
+	}
```

Comment:
How confident are you that this is correct for the general case, rather than working for the HumanFemale? Reading through the [code for `UsdSkelBakeSkinning`](https://github.com/PixarAnimationStudios/USD/blob/d8a405a1344480f859f025c4f97085143efacb53/pxr/usd/usdSkel/bakeSkinning.cpp#L1480-L1498) implies that it's the skel-to-to-prim transform we should be using, rather than the bind transform.

Reply:
I am not confident at all, though I was under the impression the geomBindTransform is moving a prim from local to skel space for binding, so taking its inverse to get back to prim space made sense to me.

Do you know how I would get the `skelLocalToWorld` and `gprimLocalToWorld` transforms to test with?

Reply:
`UsdSkelBakeSkinning` is using [`UsdGeomXformCache`](https://graphics.pixar.com/usd/docs/api/class_usd_geom_xform_cache.html) to get those matrices, but its not thread-safe... I've punted on the issue with extra comments in 0751acb1f161fb7ce88f9cf6f061af5b81986508. I think that might be most pragmatic until we have a need to make our code more complex, unless you know of a thread-safe & simple(ish) way to get the necessary matrices.

Reply:
Fine with me.